granulate_utils/linux/elf.py

```diff
@@ -12,6 +13,20 @@
 __all__ = ["ELFError"]
 
 
+def catch_filenotfound(func):
```

Comment:
Make sure we don't mess up the typing of the wrapped function. You can follow this guide: https://rednafi.github.io/reflections/static-typing-python-decorators.html, it looks okay

Reply:
Done